src/lib/consensus/proof_of_stake.ml

```diff
@@ -806,16 +824,175 @@ module Consensus_transition_data = struct
       ~var_of_hlist:of_hlist ~value_to_hlist:to_hlist ~value_of_hlist:of_hlist
 end
 
+module Global_slot = struct
+  open Snark_params.Tick
+
+  include Nat.Make32 ()
+
+  let length_in_bits = 32
+
+  (* Make sure this optimization makes sense versus using
+     an (epoch, slot) pair *)
+  let () =
+    assert (
+      Core.Int.(
+        1 + length_in_triples
+        (* Cost of creating/unpacking *)
+        + (5 * length_in_triples)
+        (* Cost of hashing *)
+        < 5 * (Epoch.length_in_triples + Epoch.Slot.length_in_triples)) )
+
+  let create ~(epoch : Epoch.t) ~(slot : Epoch.Slot.t) =
+    of_int
+      ( Epoch.Slot.to_int slot
+      + (UInt32.to_int Constants.Epoch.size * Epoch.to_int epoch) )
+
+  module Checked = struct
+    (* TODO: It's possible to share this hash computation with 
```

Comment:
what hash computation?

Reply:
I believe @imeckler is implying that we may want to add the global_slot to the state so that it is hashed along with the state hash, though I'm not 100% sure.